extensions/bitwarden/src/search.tsx

```diff
@@ -0,0 +1,191 @@
+import { dirname } from "path";
+
+const { cliPath, clientId, clientSecret, fetchFavicons } = getPreferenceValues();
+process.env.PATH = dirname(cliPath)
+const bitwardenApi = new BitwardenApi(clientId, clientSecret);
+
+export default function ListCommand(): JSX.Element {
+  if (!existsSync(cliPath)) {
+    return <TroubleshootingGuide />;
+  }
+
+  const [sessionToken, setSessionToken] = useSessionToken(bitwardenApi);
+
+  if (sessionToken === null) return <UnlockForm setSessionToken={setSessionToken} bitwardenApi={bitwardenApi} />;
+
+  return <ItemList bitwardenApi={bitwardenApi} sessionToken={sessionToken}/>;
+}
+
+function ItemList(props: { bitwardenApi: BitwardenApi; sessionToken: string | undefined }) {
+  const { bitwardenApi, sessionToken } = props;
+  const [state, setState] = useState<{ folders: Folder[]; items: Item[] }>();
+
+  const folderMap: Record<string, Folder> = {};
+  for (const folder of state?.folders || []) {
+    if (folder.id) folderMap[folder.id] = folder;
+  }
+
+  async function loadItems(sessionToken: string) {
+    try {
+      const [items, folders] = await Promise.all([
+        bitwardenApi.listItems<Item>("items", sessionToken),
+        bitwardenApi.listItems<Folder>("folders", sessionToken),
+      ]);
+
+      setState({ items, folders });
+    } catch (error) {
+      showToast(ToastStyle.Failure, "Failed to search vault", "Vault is locked");
+    }
+  }
+
+  useEffect(() => {
+    if (sessionToken) loadItems(sessionToken);
+  }, [sessionToken]);
+
+  return (
+    <List isLoading={typeof state === "undefined"}>
+      {state?.items.map((item) => (
+        <ItemListItem
+          key={item.id}
+          item={item}
+          folder={item.folderId ? folderMap[item.folderId] : undefined}
+          refreshItems={async () => {
+            if (sessionToken) {
+              const toast = await showToast(ToastStyle.Animated, "Syncing Items...");
+              await bitwardenApi.syncItems(sessionToken)
+              await loadItems(sessionToken);
+              await toast.hide();
+            }
+          }}
+        />
+      ))}
+    </List>
+  );
+}
+
+function ItemListItem(props: { item: Item; folder: Folder | undefined; refreshItems?: () => void }) {
+  const { item, folder, refreshItems } = props;
+  const { type, name, notes, identity, login, secureNote, fields, passwordHistory, card } = item;
+  const accessoryIcons = [];
+  if (folder) accessoryIcons.push(`📂 ${folder.name}`);
+  if (item.favorite) accessoryIcons.push(`⭐️`);
+
+  const fieldMap = Object.fromEntries(fields?.map((field) => [field.name, field.value]) || []);
+  const uriMap = Object.fromEntries(
+    login?.uris?.filter((uri) => uri.uri).map((uri, index) => [`uri${index + 1}`, uri.uri]) || []
+  );
+
+  let icon: string | Icon | undefined;
+  if (fetchFavicons && login?.uris?.[0]?.uri?.startsWith("https"))
+    icon = `https://s2.googleusercontent.com/s2/favicons?domain_url=${login?.uris?.[0].uri}`;
```

Comment:
```suggestion
  const URI = login?.uris?.[0]?.uri
  if (fetchFavicons && URI?.startsWith("https"))
    icon = `https://s2.googleusercontent.com/s2/favicons?domain_url=${URI}`;
```

Also, do we want to enforce that we only use this fetch method when the URI starts with `https`? What about `http`? What about valid entries that omit the protocol entirely? If the `googleusercontent` service requires a FQDN in the `domain_url` parameter, it would be best to create a `new URL()` from the value first.

Reply:
Good point.